### ISSUE TITLE
`X(x: some(X(x: none)))` is encoded as `X(x: none)`.

### ISSUE THREAD

**Issue by kayabaNerve** [kayabaNerve/nim-protobuf-serialization#16]
Because the last struct is empty, the entire structure is omitted, causing it to be marked as empty.

This is arguably desired behavior. That said, if we do encode an empty buffer at these locations, which is technically correct, just arguably pointless (hence why this original behavior may be desirable), we can ensure corrected nested object encoding an use a pure stream for the entire write, instead of substreams.

**Comment by kayabaNerve** [comment]
Due to the speed improvements and technical accuracy, I'm marking this as a bug.

**Comment by kayabaNerve** [comment]
This should be fixed in the new writer, although I didn't gain the speed improvements I wanted. To do that, you'd need to access `^bytes.len` in order to write `bytes.len`. I have yet to verify this can be done, properly, with the faststreams API.

**Comment by kayabaNerve** [comment]
`some(X(y: none()))` is now encoded as `none(X)`, but any nesting works fine. The problem is solely with the first object when you define it as an Optional type. Because the object itself is not considered optional, and `Option[X}` is encoded the same as `X`, it parses out nothing and assumes nothing means nothing.

The only thing we can do is:
- Require one field in a some() root type is encoded. This will cause `some(X(y: some(Y())))` which is invalid.
- Write Optional buffers as Length Delimited objects. This is a horrible idea as it makes Options a first-class type in Protobuf, not a first-class type in Nim mapped to a type in Protobuf.

**Comment by kayabaNerve** [comment]
TL;DR specifying a blank object 5 level deeps surrounded by 4 levels of objects blank except for the next level works perfectly. Specifying a blank object at the initial level, even if it does 'exist', will be treated as non-existent. Using `.get()` with a default value, or just not reading the object with the `Option` generic, works perfectly.
